sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+                    urlConnection.setRequestProperty("Content-Length", "" + contentLength);
+                    printout = new DataOutputStream(urlConnection.getOutputStream());
+                }
+                // Write the string in UTF-8 encoding
+                printout.write(content.toString().getBytes("UTF-8"));
+                printout.flush();
+                printout.close();
+            }
+
+            // get response, or read error message
+            InputStream inputStream;
+            try {
+                inputStream = urlConnection.getInputStream();
+            } catch (IOException e) {
+                inputStream = urlConnection.getErrorStream();
+            }
+            if (null == inputStream) {
+                throw new IOException("could not get InputStream");
+            }
+            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream), 1024);
+            String line;
+            StringBuffer responseContent = new StringBuffer();
+            while ((line = reader.readLine()) != null) {
+                responseContent.append(line);
+                responseContent.append('\r');
+            }
+            result.put(RESPONSE_CONTENT, responseContent.toString());
+            result.put(RESPONSE_CODE, Integer.toString(urlConnection.getResponseCode()));
+
+            // clean up
+            reader.close();
+            reader = null;
+            inputStream.close();
+            inputStream = null;
+
+            // get headers
+            Map<String, List<String>> headerFields = urlConnection.getHeaderFields();
+            String key, valueString;
+            List<String> value;
+            for (Map.Entry<String, List<String>> entry : headerFields.entrySet()) {
+                key = entry.getKey();
+                value = entry.getValue();
+                if (null != key && null != value) {
+                    key = key.toLowerCase(Locale.ENGLISH);
+                    valueString = value.toString();
+                    valueString = valueString.substring(1, valueString.length() - 1);
+                    result.put(key, valueString);
+                }
+            }
+
+            return result;
+
+        } finally {
+            if (urlConnection != null) {
+                urlConnection.disconnect();
+            }
+        }
+    }
+
+    /**
+     * Upload sensor data to commonsense
+     *
+     * Data can be coming from multiple sensors.
+     *
+     * @param responseCode	The response code returned from the request
+     * @param method		The method that requires code checking
+     *
+     * @result	The integer value of the result
+     *
+     */
+    private static int checkResponseCode(String responseCode, String method){
+        if ("403".equalsIgnoreCase(responseCode)) {
+            Log.w(TAG, "CommonSense" + method + "refused! Response: forbidden!");
+            return -3;
+        } else if ("201".equalsIgnoreCase(responseCode)) {
```

Comment:
@excaliburnan not really a problem, but for clarity I would use 0 and positive values for positive results 201 would then be 1